libc/cmake/modules/LLVMLibCCompileOptionRules.cmake

```diff
@@ -43,6 +43,7 @@ function(_get_common_compile_options output_var flags)
     list(APPEND compile_options "-fpie")
 
     if(LLVM_LIBC_FULL_BUILD)
+      list(APPEND compile_options "-DLIBC_FULL_BUILD")
```

Comment:
do we want the default state if the user doesn't set any build flags to be fullbuild or overlay?

Reply:
Currently, the default state is overlay mode, and probably we won't be able to change the default until our implementation is complete enough, and most of our downstream users switch to full build mode.

Reply:
In cmake I agree that leaving the default as overlay mode for now makes sense, but in the code I think if no compile flags are set it should default to fullbuild. My guess is that once we make this choice we won't want to switch later, and I think people creating their own build systems will want fullbuild eventually.

Reply:
It's possibly still worth having this conversation, but the diff this thread is off of was made obsolete by #87598.  File a new bug if you'd like to discuss this more?